### PR TITLE
Updated kotlinVersion to 1.2.40

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-api/build.gradle.kts [hunk 1]
```diff
@@ -13,4 +13,5 @@ dependencies {
 	testRuntimeOnly("org.junit.platform:junit-platform-launcher:$junitPlatformVersion")
 	testRuntimeOnly("org.junit.platform:junit-platform-console:$junitPlatformVersion")
 	testRuntimeOnly("org.jetbrains.spek:spek-junit-platform-engine:$spekVersion")
+	compile("org.jetbrains.kotlin:kotlin-reflect")
```
[the remark this answers is not in the document]
Comment (schalkms) [reply detekt-api/build.gradle.kts]: We don't need the `Reflective` interface.

**Comment by schalkms** [comment]
@arturbosch please check out this branch before merging.

**Comment by arturbosch** [comment]
@schalkms it builds and runs on all analysis-projects. Merge? :)

**Comment by schalkms** [comment]
@arturbosch yes!

**Pull request merged**